### PR TITLE
fix: resolve modules if folder contains a package.json file

### PR DESCRIPTION
**Pull request by scott-ut**
getMappedPath resolves files by trying a number of different file extensions and filtering out those that aren't a file. This does not work when a TSConfig path alias resolves to a _directory_ containing built assets, and described by a package.json, e.g. `@foo/bar` -> `node_modules/foo/bar/dist`.

If the tsconfig path is mapped to a folder that doesn't contain an `index.${ext}` file, then getMappedPath returns undefined as the folder itself is not a file. However, in some cases it is ok for this to resolve to the directory as the package.json contains a `typings` entry that references a .d.ts file, which is correctly handled by enhanced-resolve.

https://github.com/import-js/eslint-import-resolver-typescript/issues/188

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -250,6 +250,10 @@ const isFile = (path?: string | undefined): path is string => {
   }
 }
 
+const isModule = (path?: string | undefined): path is string => {
+  return !!path && isFile(`${path}/package.json`)
```
Comment (JounQin) [inline src/index.ts:254]: It'll fail on Windows.

**Review by JounQin — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (scott-ut) [reply src/index.ts]: Changed this to use `path.sep` instead of `/`

File: src/index.ts [hunk 2]
```diff
@@ -250,6 +250,10 @@ const isFile = (path?: string | undefined): path is string => {
   }
 }
 
+const isModule = (modulePath?: string | undefined): modulePath is string => {
+  return !!modulePath && isFile(`${modulePath}${path.sep}package.json`)
```
Comment (JounQin) [inline src/index.ts:254]:
```suggestion
  return !!modulePath && isFile(path.resolve(modulePath, 'package.json')
```

**Review by JounQin — APPROVED** [review]

**Comment by JounQin** [comment]
@scott-ut Thanks for your contribution!

**Pull request merged**